src/rest_vol.c

```diff
@@ -2219,9 +2266,14 @@ RV_find_object_by_path(RV_object_t *parent_obj, const char *obj_path,
 #endif
 
         if (ret_value > 0) {
-            if (obj_found_callback && RV_parse_response(response_buffer.buffer,
-                    callback_data_in, callback_data_out, obj_found_callback) < 0)
+            if (obj_found_callback == RV_copy_object_URI_and_domain_callback) {
```

Comment:
Rather than changing this line to look for a specific callback, the code calling this function should be aware that it needs to pass `parent_obj->domain` for `callback_data_in` and those places should be changed instead.

Reply:
In the case of an external link, `parent_obj` will have been reassigned to be an external file at [line 2043](https://github.com/HDFGroup/vol-rest/blob/master/src/rest_vol.c#L2043), so passing `parent_obj->domain` from the code calling this function would always provide the original, potentially incorrect domain.

Changing `callback_data_in` when an external link is found wouldn't be a solution either, as that would break any callbacks which use `callback_data_in` when they're used on an external link.

Reply:
This seems like an architectural issue. Relying on RV_find_object_by_path to check for a specific callback function and then replace the passed in data with internal data is a fragile solution that will likely lead to bugs if other similar callback functions are added in the future.

Reply:
I was able to resolve this - the server returns the URI and path to the domain for the object, which can be used to determine if an object is external and to construct a new domain object.